File: Python/EstruturaDeRepeticao/exercicio_18.py
"""
Faça um programa que, dado um conjunto de N números, determine o menor valor, o maior valor e a soma dos valores.
"""

conjunto = int(input("Digite a quantidade de números no conjunto: "))

numeros = []
for i in range(1, conjunto+1):
    numero = int(input(f"Digite o {i}º número: "))
    numeros.append(numero)

maior = max(numeros)
menor = min(numeros)
soma = sum(numeros)

print(f"Menor valor: {menor}")
print(f"Maior valor: {maior}")
print(f"Soma dos valores: {soma}")

File: Java/EstruturaDeRepeticao/Exercicio18.java
// Faça um programa que, dado um conjunto de N números, determine o menor valor, o maior valor e a soma dos valores.
package EstruturaDeRepeticao;
import java.util.Scanner;

public class Exercicio18 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Digite a quantidade de números (N): ");
        int N = scanner.nextInt();

        int menor = Integer.MAX_VALUE;
        int maior = Integer.MIN_VALUE;
        int soma = 0;

        for (int i = 1; i <= N; i++) {
            System.out.print("Digite o " + i + "º número: ");
            int numero = scanner.nextInt();

            if (numero < menor) {
                menor = numero;
            }

            if (numero > maior) {
                maior = numero;
            }

            soma += numero;
        }

        System.out.println("Menor valor: " + menor);
        System.out.println("Maior valor: " + maior);
        System.out.println("Soma dos valores: " + soma);

        scanner.close();
    }
}
